nixos/modules/services/databases/victoriatraces.nix

```diff
@@ -0,0 +1,173 @@
+{
+  config,
+  pkgs,
+  lib,
+  utils,
+  ...
+}:
+let
+  inherit (lib)
+    escapeShellArgs
+    getBin
+    hasPrefix
+    literalExpression
+    mkBefore
+    mkEnableOption
+    mkIf
+    mkOption
+    mkPackageOption
+    optionalString
+    types
+    ;
+  cfg = config.services.victoriatraces;
+  startCLIList = [
+    "${cfg.package}/bin/victoria-traces"
+    "-storageDataPath=/var/lib/${cfg.stateDir}"
+    "-httpListenAddr=${cfg.listenAddress}"
+  ]
+  ++ lib.optionals (cfg.basicAuthUsername != null) [
+    "-httpAuth.username=${cfg.basicAuthUsername}"
+  ]
+  ++ lib.optionals (cfg.basicAuthPasswordFile != null) [
+    "-httpAuth.password=file://%d/basic_auth_password"
+  ];
+in
+{
+  options.services.victoriatraces = {
+    enable = mkEnableOption "VictoriaTraces is an open source distributed traces storage and query engine from VictoriaMetrics";
+    package = mkPackageOption pkgs "victoriatraces" { };
+    listenAddress = mkOption {
```

Comment:
i think this should be spil into a `listenAddress` and a port option. Because usually people dont talk about TCP addresses and the port being distinct from the address/interface being listened on (and there even is a `lib.types.port` now).